app/models/search.server.ts

```diff
@@ -42,6 +42,23 @@ function calculateElementsRemaining(remaining: number, gotten: number) {
   return remaining - gotten
 }
 
+export async function getSearchResults(
+  text: string,
+  page: string = "1",
+  caseSensitive: boolean = false,
+  attribute: string = SearchResultEnum.HEADWORD
+) {
+  switch (attribute) {
+    case SearchResultEnum.HEADWORD:
+      return getEntriesByBasicTextSearchAndPage(text, page, caseSensitive)
+    case SearchResultEnum.MEANING:
+      return getSearchResultsFromMeaningsAndPage(text, page, caseSensitive)
+    default:
+      throw new Error(`attribute ${attribute} must be a valid search result`)
+  }
+}
+
+// TODO: Delete this function and related
 export async function getSearchResultsByPage(
   text: string,
   page: string = "1",
```

Comment:
this is the old function. I'm gonna delete this in a future PR just to keep this from getting too crowded